test/common/http/conn_manager_impl_test.cc

```diff
@@ -385,6 +387,10 @@ TEST_F(HttpConnectionManagerImplTest, StartAndFinishSpanNormalFlow) {
         callbacks.addStreamDecoderFilter(filter);
       }));
 
+  // Treat request as internal, otherwise x-request-id header will be overwritten.
```

Comment:
Q: Are all of this similar changes necessary? Just wondering why they had to be made?